project/src/test/java/com/google/sps/data/PlacesFetcherTest.java

```diff
@@ -125,13 +131,21 @@ public void fetch_zeroSearchResults_returnsEmptyList() throws Exception {
   }
 
   @Test
-  public void fetch_validSearchResults_returnsListOfPlaces()
-      throws ApiException, InterruptedException, IOException {
+  public void fetch_validSearchResults_returnsListOfPlaces() throws Exception {
     PlacesFetcher spiedFetcher = spy(placesFetcher);
     doReturn(PLACE_DETAILS_1).doReturn(PLACE_DETAILS_2)
         .when(spiedFetcher).getPlaceDetails(any(PlaceDetailsRequest.class));
     doReturn(SEARCH_RESULT_ARR)
         .when(spiedFetcher).getPlacesSearchResults(any(TextSearchRequest.class));
     assertEquals(ImmutableList.of(PLACE_1, PLACE_2), spiedFetcher.fetch());
   }
+
+  @Test
+  public void fetch_PlaceDetailsQueryFails_throwsFetcherException()
+      throws FetcherException {
```

Comment:
I think you should add `throws Exception` to every test and then not bother to know which exception might get thrown.

Reply:
Agree